pkg/httpclientutil/httpclientutil_others.go

```diff
@@ -17,9 +19,8 @@ func NewHTTPClientWithSocketPath(socketPath string) (*http.Client, error) {
 	}
 	hc := &http.Client{
 		Transport: &http.Transport{
-			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
-				var d net.Dialer
-				return d.DialContext(ctx, "unix", socketPath)
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return netutil.DialUnix("unix", socketPath)
```

Comment:
Here we ignore the context, which may ignore a deadline no? I would keep the code as is for now and focus on the places we use net.DialUnix(). Keeping the change small helps to review it faster is easy to merge faster.